src/chat/chat_tests.rs

```diff
@@ -2627,6 +2627,121 @@ async fn test_can_send_group() -> Result<()> {
     Ok(())
 }
 
+/// Tests that in a broadcast channel,
+/// the recipients can't see the identity of their fellow recipients.
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_broadcast_members_cant_see_each_other() -> Result<()> {
+    fn contains(parsed: &MimeMessage, s: &str) -> bool {
+        assert_eq!(parsed.decrypting_failed, false);
+        let decoded_str = str::from_utf8(&parsed.decoded_data).unwrap();
+        decoded_str.contains(s)
+    }
+
+    let mut tcm = TestContextManager::new();
+    let alice = &tcm.alice().await;
+    let bob = &tcm.bob().await;
+    let charlie = &tcm.charlie().await;
+
+    tcm.section("Alice creates a channel, Bob joins.");
+    let alice_broadcast_id = create_broadcast(alice, "Channel".to_string()).await?;
+    let qr = get_securejoin_qr(alice, Some(alice_broadcast_id))
+        .await
+        .unwrap();
+    tcm.exec_securejoin_qr(bob, alice, &qr).await;
+
+    tcm.section("Charlie scans the QR code and sends request.");
+    {
+        join_securejoin(charlie, &qr).await.unwrap();
+
+        let request = charlie.pop_sent_msg().await;
+        assert_eq!(request.recipients, "alice@example.org charlie@example.net");
+
+        alice.recv_msg_trash(&request).await;
+    }
+
+    tcm.section("Alice sends auth-required");
+    {
+        let auth_required = alice.pop_sent_msg().await;
+        assert_eq!(
+            auth_required.recipients,
+            "charlie@example.net alice@example.org"
+        );
+        let parsed = charlie.parse_msg(&auth_required).await;
+        assert!(parsed.header_exists(HeaderDef::AutocryptGossip));
+        assert!(contains(&parsed, "charlie@example.net"));
+        assert_eq!(contains(&parsed, "bob@example.net"), false);
```

Comment:
`assert!(contains(&parsed, "bob@example.net") == false);` reads better to me (and is better aligned with the assert-s above)